source/common/http/async_client_impl.h

```diff
@@ -80,6 +80,8 @@ class AsyncStreamImpl : public AsyncClient::Stream,
   AsyncStreamImpl(AsyncClientImpl& parent, AsyncClient::StreamCallbacks& callbacks,
                   const AsyncClient::StreamOptions& options);
 
+  bool requestRouteConfigUpdate(std::function<void()>) override { return false; }
```

Comment:
Wait, is this necessary? Could you put a `NOT_IMPLEMENTED_GCOVR_EXCL_LINE` in the interface header, like you have in subscription.h, and just not override it here?

Reply:
I I changed the interface (added canRequestRouteConfigUpdate check), and updated to use NOT_IMPLEMENTED_GCOVR_EXCL_LINE where possible. Please let me know what you think

Reply:
Looks good, makes sense.